internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -501,6 +501,63 @@ ExecExprVisitor::ExecUnaryRangeVisitorDispatcher(UnaryRangeExpr& expr_raw)
     return ExecUnaryRangeVisitorDispatcherImpl<T>(expr_raw);
 }
 
+template <typename T>
+bool
+compareTwoJsonArray(T arr1, const proto::plan::Array& arr2) {
```

Comment:
Please add static assertion for T, so we can remember to add logic when new type is supported.